src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotLockOnObjectsWithWeakIdentity.cs

```diff
@@ -51,10 +51,18 @@ public override void Initialize(AnalysisContext analysisContext)
                 compilationStartContext.RegisterOperationAction(context =>
                 {
                     var lockStatement = (ILockOperation)context.Operation;
-                    ITypeSymbol type = lockStatement.LockedValue?.Type;
-                    if (type != null && TypeHasWeakIdentity(type, compilation))
+
+                    if (lockStatement.LockedValue.Kind == OperationKind.InstanceReference)
```

Comment:
I decided not to go for a separate registered action (even if I know syntax access is faster) mainly because this project doesn't reference Visual Basic syntax kinds so I won't be able to also test for the `Me` syntax.

Reply:
Can you make the check even more explicit/strict here by validating the `ReferenceKind` of the instance reference?

```csharp
if (lockStatement.LockedValue is IInstanceReferenceOperation instanceReference &&
    instanceReference.ReferenceKind == InstanceReferenceKind.ContainingTypeInstance)
```